config/external.go

```diff
@@ -42,6 +43,25 @@ type External struct {
 	// the provider. If specified, it will take precedence over the "garm-external-provider"
 	// executable in the ProviderDir.
 	ProviderExecutable string `toml:"provider_executable" json:"provider-executable"`
+	// EnvironmentVariables is a map of environment variable names that will be
+	// passed to the external binary together with their values.
+	EnvironmentVariables []string `toml:"environment_variables" json:"environment-variables"`
+}
+
+func (e *External) GetEnvironmentVariables() []string {
+	envVars := []string{}
+
+	for _, configuredEnvVars := range e.EnvironmentVariables {
+		// discover environment variables
+		for _, k := range os.Environ() {
+			variable := strings.SplitN(k, "=", 2)
+
```

Comment:
We should probably skip any pattern that starts with `GARM`. We don't want to override the ones that are set by garm. See bellow comment.